code/game/objects/items/devices/scanners.dm

```diff
@@ -693,12 +693,59 @@ GENE SCANNER
 		else
 			to_chat(user, "Possible mutations: [T.slime_mutation[1]], [T.slime_mutation[2]], [T.slime_mutation[3]], [T.slime_mutation[4]]")
 			to_chat(user, "Genetic destability: [T.mutation_chance] % chance of mutation on splitting")
-	if (T.cores > 1)
+	if(T.cores > 1)
 		to_chat(user, "Multiple cores detected")
 	to_chat(user, "Growth progress: [T.amount_grown]/[SLIME_EVOLUTION_THRESHOLD]")
 	if(T.effectmod)
 		to_chat(user, "<span class='notice'>Core mutation in progress: [T.effectmod]</span>")
 		to_chat(user, "<span class = 'notice'>Progress in core mutation: [T.applied] / [SLIME_EXTRACT_CROSSING_REQUIRED]</span>")
+	if(T.transformeffects != SLIME_EFFECT_DEFAULT)
+		var/slimeeffect = "\nTransformative extract effect detected: "
+		if(T.transformeffects & SLIME_EFFECT_GREY)
+			slimeeffect += "grey"
+		if(T.transformeffects & SLIME_EFFECT_ORANGE)
+			slimeeffect += "orange"
+		if(T.transformeffects & SLIME_EFFECT_PURPLE)
+			slimeeffect += "purple"
+		if(T.transformeffects & SLIME_EFFECT_BLUE)
+			slimeeffect += "blue"
+		if(T.transformeffects & SLIME_EFFECT_METAL)
+			slimeeffect += "metal"
+		if(T.transformeffects & SLIME_EFFECT_YELLOW)
+			slimeeffect += "yellow"
+		if(T.transformeffects & SLIME_EFFECT_DARK_PURPLE)
+			slimeeffect += "dark purple"
+		if(T.transformeffects & SLIME_EFFECT_DARK_BLUE)
+			slimeeffect += "dark blue"
+		if(T.transformeffects & SLIME_EFFECT_SILVER)
+			slimeeffect += "silver"
+		if(T.transformeffects & SLIME_EFFECT_BLUESPACE)
+			slimeeffect += "bluespace"
+		if(T.transformeffects & SLIME_EFFECT_SEPIA)
+			slimeeffect += "sepia"
+		if(T.transformeffects & SLIME_EFFECT_CERULEAN)
+			slimeeffect += "cerulean"
+		if(T.transformeffects & SLIME_EFFECT_PYRITE)
+			slimeeffect += "pyrite"
+		if(T.transformeffects & SLIME_EFFECT_RED)
+			slimeeffect += "red"
+		if(T.transformeffects & SLIME_EFFECT_GREEN)
+			slimeeffect += "green"
+		if(T.transformeffects & SLIME_EFFECT_PINK)
+			slimeeffect += "pink"
+		if(T.transformeffects & SLIME_EFFECT_GOLD)
+			slimeeffect += "gold"
+		if(T.transformeffects & SLIME_EFFECT_OIL)
+			slimeeffect += "oil"
+		if(T.transformeffects & SLIME_EFFECT_BLACK)
+			slimeeffect += "black"
+		if(T.transformeffects & SLIME_EFFECT_LIGHT_PINK)
+			slimeeffect += "light pink"
+		if(T.transformeffects & SLIME_EFFECT_ADAMANTINE)
+			slimeeffect += "adamantine"
+		if(T.transformeffects & SLIME_EFFECT_RAINBOW)
+			slimeeffect += "rainbow"
+		to_chat(user, "<span class='notice'>[slimeeffect].</span>")
```

Comment:
now this just shows as slime effects detector: greyredbluerainbow
you need something like
var/list/effects = list()
effects += "black"
effects += "rainbow"]
if(effects.len)
   to_chat(user, "Slime effects detected: [effects.join(", ")]")

Reply:
a slime can only have one effect at a time, I don't see the problem.

Reply:
> now this just shows as slime effects detector: greyredbluerainbow

Where did you find out what is wrong?

Reply:
I mirror this - where are you getting this error? When I examine slimes with the console it always only shows me one color no matter how many times I've changed the color.

Reply:
> a slime can only have one effect at a time

Reply:
I didn't know that

Reply:
But why is it using bit flags if you can only have 1 effect at once?

Reply:
initially I wanted to facilitate the work of a possible rework where someone wants to make stackable effects.

Reply:
So, are you gonna redo it the way Bacon said? I'd love to see this merged it's super cool.

Reply:
> So, are you gonna redo it the way Bacon said?
no

Reply:
That's too bad. Thanks for making it, sadly don't think it'll get merged in that case.

Reply:
> That's too bad. Thanks for making it, sadly don't think it'll get merged in that case.

I don't see why it shouldn't be marginalized. Your suggestion was made by having the wrong thought about what should happen.

He made that suggestion to correct an error but the error never existed.

Reply:
Ah ok my bad.